### PR TITLE
feat: add `external-auth` cli

### PR DESCRIPTION
**Pull request by kylecarbs**
This will allow us to enable scripting around external authentication. I assume we'll add more functionality to the `external-auth` subcommand, which is why `access-token` is nested.

```hcl
data "coder_external_auth" "slack" {
  id = "slack"
}

resource "coder_script" "notify-slack" {
  # Write a file to the $PATH that executes `coder external-auth access-token slack` at the
  # beginning to check for permissions, awaits for a command to execute and notifies the user
  # via a Slack message!
}
```

### CODE REVIEW

**Comment by Shocktrooper** [comment]
One thing I noticed with the external auth is since open source is limited to 1 external auth provider it makes it hard to test out a git provider + an oidc application together to see if they work for testing templates(our dev environment uses Open Source).  Would it be possible to bump the open source limit to 1 git provider + 1 alternate provider configuration that is not a git based provider?

**Comment by kylecarbs** [comment]
@Shocktrooper could you share more about the specific use-case you're solving for? I'm open to the idea, I just wanna make sure it's not something that could be discovered in a free trial.

File: cli/externalauth.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+			"If a valid access-token cannot be obtained, the URL to authenticate will be sent to stdout with exit code 1\n" + formatExamples(
+			example{
+				Description: "Ensure that the user is authenticated with GitHub before cloning.",
+				Command: `#!/usr/bin/env sh
+
+if coder external-auth access-token github ; then
+  echo "Authenticated with GitHub"
+else
+  echo "Please authenticate with GitHub:"
+  coder external-auth url github
```
Comment (coadler) [inline cli/externalauth.go:43]: I don't see a `external-auth url` cmd.

**Review by coadler — APPROVED** [review]

File: cli/externalauth.go (hunk 1 above)
Comment (kylecarbs) [reply cli/externalauth.go:43]: Good find!

**Pull request merged**